server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -468,6 +551,13 @@ public String getDefaultCodec() {
         return CodecService.DEFAULT_CODEC;
     }
 
+    /**
+     * The default index settings provided by this index mode.
+     */
+    public Settings defaultIndexSettings() {
```

Comment:
Like we discussed yesterday. I think instead of adding defaultIndexSettings() method to index mode and then invoking that from `MetadataCreateIndexService`, `validateWithOtherSettings` and `MetadataIndexTemplateService` (currently missing in this change), we can implement this via an index settings provider (`IndexSettingProvider`)? If index mode is lookup, then it would return a settings instance with `index.number_of_shards` and `index.auto_expand_replicas` settings set to `1` and `0-all` respectively.  The validation in `LOOKUP.validateWithOtherSettings(...)` should then remain.